packages/react/src/themes/base/components/Text/textVariables.ts

```diff
@@ -1,8 +1,7 @@
-import { ColorValues } from '../../../types'
```

Comment:
could you, please, walk me through client's workflow on this. So, I would like to introduce `Foo` component that is able to color itself with theme being switched. What is the source I need to use colors from? 

Am I right that I am supposed to take a look on the Design Spec and use colors from `siteVariables`, as a first thing? Like 

```js
// variables, default theme
return {
    atMentionMeColor: siteVariables.colors.pink[500], // those are palette colors
    atMentionOtherColor: siteVariables.colors.grey[600],
    importantColor: siteVariables.colors.red[500]
   ...
}
```

Then, to override colors for different themes, I need to override corresponding variables, according to the Design Spec:
```js
// variables, Dark Theme
return {
    atMentionMeColor: siteVariables.colors.red[500], // overriding with palette color
   ...
}
```

And then, if I would like to support color variants for my component (i.e. `color` prop), then this is the moment when I need to plug `colorScheme`, right - otherwise I don't need it?

```js
// variables, default theme
return {
    colorScheme: siteVariables.colorScheme,
    atMentionMeColor: siteVariables.colors.pink[500], // those are palette colors
    atMentionOtherColor: siteVariables.colors.grey[600],
    importantColor: siteVariables.colors.red[500]
   ...
}
```

Reply:
Even if the component doesn't support color, the color scheme defines the colors that should be used across the app. Design should specify in the specs which colors should be used for this component, for example:

```
// variables, default theme
return {
    atMentionMeColor: siteVariables.colorScheme.pink.foreground, // those are palette colors
    atMentionOtherColor: siteVariables.colorScheme.default.foreground1,
    importantColor: siteVariables.colorScheme.red.foreground,
   ...
}
```
This colors will be mapped correctly in all different themes, so the user won't have to do the overrides. Currently we have only the default and primary color schemes, but design is working on defining the schemes for the other colors as well.

On the other hand, if the component supports the color prop, you are right, they can leverage further the color scheme, by picking the color provided by the color prop.

Reply:
got it - so, essentially, the vector of Design Spec rules reduces to the following:

- wherever it is possible, use colors from color sheme
- if rules of color scheme doesn't apply, define color from color palatte

Am I right here?

Reply:
> wherever it is possible, use colors from color sheme

Yep, totally right
> if rules of color scheme doesn't apply, define color from color palatte

Yep, and override them to the correct values in all different themes.